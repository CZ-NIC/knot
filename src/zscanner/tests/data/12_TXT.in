$ORIGIN	.
$TTL	1

; OK
@	TXT	""				; Blank string
@	TXT	a				; One char string
@	TXT	\ 				; One space char
@	TXT	"\ "				; One space char in quotes
@	TXT	\021				; One unprintable char
@	TXT	"\\ \""				; Special chars
@	TXT	"" "test1" "\255" test2		; Array of text strings
@	TXT	"" "" ""			; Array of blank strings
@	TXT	first \# "\#"			; Array with special string
@	TXT	\0320\ \\\"\.\@\*.tld.		; Special domain as a string
@	TXT	" !\"#$%&'()*+,-./0123456789:;<=>?@"	; First part of all printables
@	TXT	"ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`"	; Second part of all printables
@	TXT	"abcdefghijklmnopqrstuvwxyz{|}~"	; Third part of all printables
@	TXT	\# 1 00				; Hexadecimal rdata
@	TYPE16	\# 1 00				; TYPE + Hexadecimal rdata
@	TYPE16	""				; TYPE
@	TXT	(				; Special multi-line string
"first
second"
third						; Second string
)
@	TXT	"abcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMNabcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMNabcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMNabcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMNabcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMN1234\053"	; Text string of maximal length (255 chars)
@	TXT	"abcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMNabcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMNabcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMNabcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMNabcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMN1234\0536"	; Minimum overflowed text string which should be divided into two strings.
@	TXT	"abcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMNabcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMNabcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMNabcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMNabcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMN1234\053\054"	; Minimum overflowed text string which should be divided into two strings (decimal version).
@	txt	""				; Type in lower-case

; KO
@	TXT
@	TXT			; Empty rdata
@	TXT	\# 0		; Hex empty rdata
@	TXT	\01		; Missing digit in decimal notation
@	TXT	\256		; 8bit overflow in decimal notation
@	TXT	"""		; '"' char without forward slash
