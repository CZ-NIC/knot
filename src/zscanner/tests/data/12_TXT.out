OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0010
RDATA=00
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0010
RDATA=0161
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0010
RDATA=0120
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0010
RDATA=0120
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0010
RDATA=0115
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0010
RDATA=035C2022
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0010
RDATA=0005746573743101FF057465737432
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0010
RDATA=000000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0010
RDATA=05666972737401230123
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0010
RDATA=0D2030205C222E402A2E746C642E
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0010
RDATA=21202122232425262728292A2B2C2D2E2F303132333435363738393A3B3C3D3E3F40
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0010
RDATA=204142434445464748494A4B4C4D4E4F505152535455565758595A5B5C5D5E5F60
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0010
RDATA=1E6162636465666768696A6B6C6D6E6F707172737475767778797A7B7C7D7E
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0010
RDATA=00
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0010
RDATA=00
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0010
RDATA=00
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0010
RDATA=0C66697273740A7365636F6E64057468697264
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0010
RDATA=FF6162636465666768696A6B6C6D6E6F707172737475767778797A313233343536373839304142434445464748494A4B4C4D4E6162636465666768696A6B6C6D6E6F707172737475767778797A313233343536373839304142434445464748494A4B4C4D4E6162636465666768696A6B6C6D6E6F707172737475767778797A313233343536373839304142434445464748494A4B4C4D4E6162636465666768696A6B6C6D6E6F707172737475767778797A313233343536373839304142434445464748494A4B4C4D4E6162636465666768696A6B6C6D6E6F707172737475767778797A313233343536373839304142434445464748494A4B4C4D4E3132333435
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0010
RDATA=FF6162636465666768696A6B6C6D6E6F707172737475767778797A313233343536373839304142434445464748494A4B4C4D4E6162636465666768696A6B6C6D6E6F707172737475767778797A313233343536373839304142434445464748494A4B4C4D4E6162636465666768696A6B6C6D6E6F707172737475767778797A313233343536373839304142434445464748494A4B4C4D4E6162636465666768696A6B6C6D6E6F707172737475767778797A313233343536373839304142434445464748494A4B4C4D4E6162636465666768696A6B6C6D6E6F707172737475767778797A313233343536373839304142434445464748494A4B4C4D4E31323334350136
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0010
RDATA=FF6162636465666768696A6B6C6D6E6F707172737475767778797A313233343536373839304142434445464748494A4B4C4D4E6162636465666768696A6B6C6D6E6F707172737475767778797A313233343536373839304142434445464748494A4B4C4D4E6162636465666768696A6B6C6D6E6F707172737475767778797A313233343536373839304142434445464748494A4B4C4D4E6162636465666768696A6B6C6D6E6F707172737475767778797A313233343536373839304142434445464748494A4B4C4D4E6162636465666768696A6B6C6D6E6F707172737475767778797A313233343536373839304142434445464748494A4B4C4D4E31323334350136
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0010
RDATA=00
------
WARNG=ZS_BAD_RDATA
------
WARNG=ZS_BAD_RDATA
------
WARNG=ZS_BAD_HEX_CHAR
------
WARNG=ZS_BAD_NUMBER
------
WARNG=ZS_NUMBER8_OVERFLOW
------
WARNG=ZS_BAD_TEXT
------
